hack/verify-versionize-docs.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Verifies that running versionize-docs the second time, does not result in any
+# diff.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+API_ROOT="${KUBE_ROOT}/pkg/api/"
+SWAGGER_SPEC_ROOT="${KUBE_ROOT}/api/swagger-spec/"
+DOCS_ROOT="${KUBE_ROOT}/docs/"
+EXAMPLES_ROOT="${KUBE_ROOT}/examples/"
+TMP_ORIGINAL_ROOT="/tmp/tmp-original-files/"
+TMP_VERSIONED_ROOT="/tmp/tmp-versioned-files/"
+
+# Copy the original version of all files that are going to be changed by
+# versionize-docs.
+echo "Copying original docs to ${TMP_ORIGINAL_ROOT}"
+mkdir -p "${TMP_ORIGINAL_ROOT}"
+cp -a "${DOCS_ROOT}" "${TMP_ORIGINAL_ROOT}"
+cp -a "${EXAMPLES_ROOT}" "${TMP_ORIGINAL_ROOT}"
+mkdir "${TMP_ORIGINAL_ROOT}pkg"
+cp -a "${API_ROOT}" "${TMP_ORIGINAL_ROOT}pkg"
+mkdir "${TMP_ORIGINAL_ROOT}api"
+cp -a "${SWAGGER_SPEC_ROOT}" "${TMP_ORIGINAL_ROOT}api"
+
+# Run versionize-docs.
+echo "Running versionize-docs for the first time"
+${KUBE_ROOT}/build/versionize-docs.sh v1 >/dev/null
```

Comment:
Yea that is what I tried first, but none of these scripts support an OUTPUT_DIR. Looking around, the tests for those scripts do the same thing as I have done with this one i.e copy the original files that will be mutated by the script, run the script to mutate the repo, do the comparison and then copy back the original files.
So I assumed its an accepted pattern :)

If you do want me to change the test, I think that will be a big enough change, which I will prefer to do in a separate PR. I can remove the test file from this PR for now, and send a separate PR with the tests redone? Or we can submit this as is and improve it later?